diff --git a/dist/server/base-server.js b/dist/server/base-server.js
index 591457ba2c1b77424346fa9dbad9775297111a37..f4f7d982916794c3bfe14f5d721080efdb91b5a7 100644
--- a/dist/server/base-server.js
+++ b/dist/server/base-server.js
@@ -427,6 +427,13 @@ class Server {
         _log.error(err);
     }
     async handleRequest(req, res, parsedUrl) {
+        // Append the url in the global content to allow 
+        // every component or function to access the current url at any place
+        // registered by next after this point
+        globalThis.request = {
+            url: req.url,
+            id: randomUUID()
+        }
         await this.prepare();
         const method = req.method.toUpperCase();
         const rsc = isRSCRequestCheck(req) ? "RSC " : "";
